editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/globalconfiguration/beans/wizards/EditGlobalBeanWizard.java

```diff
@@ -0,0 +1,116 @@
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.jface.wizard.Wizard;
+import org.eclipse.ui.IWorkbench;
+import org.fusesource.ide.camel.editor.component.wizard.ComponentManager;
+import org.fusesource.ide.camel.editor.globalconfiguration.beans.wizards.pages.GlobalBeanEditWizardPage;
+import org.fusesource.ide.camel.editor.provider.ext.GlobalConfigurationTypeWizard;
+import org.fusesource.ide.camel.model.service.core.catalog.components.Component;
+import org.fusesource.ide.camel.model.service.core.catalog.components.ComponentModel;
+import org.fusesource.ide.camel.model.service.core.model.CamelBean;
+import org.fusesource.ide.camel.model.service.core.model.CamelFile;
+import org.w3c.dom.Element;
+
+/**
+ * @author bfitzpat
+ *
+ */
+public class EditGlobalBeanWizard extends Wizard implements GlobalConfigurationTypeWizard {
+
+	private DataBindingContext dbc;
+	private Element inputNode;
+	private CamelFile camelFile;
+	private ComponentManager componentManager;
+
+	public EditGlobalBeanWizard(CamelFile camelFile, ComponentModel componentModel) {
+		super();
+		this.camelFile = camelFile;
+		this.dbc = new DataBindingContext();
+		this.componentManager = new ComponentManager(componentModel);
+		setWindowTitle("Edit Bean");
+		setNeedsProgressMonitor(true);
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see org.eclipse.jface.wizard.Wizard#addPages()
+	 */
+	@Override
+	public void addPages() {
+		super.addPages();
+		final GlobalBeanEditWizardPage globalBeanPage = new GlobalBeanEditWizardPage(dbc, "Bean Definition",
+				"Edit details for selected bean definition.", camelFile);
+		globalBeanPage.setElement(inputNode);
+		addPage(globalBeanPage);
+
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see org.fusesource.ide.camel.editor.provider.ext.
+	 * GlobalConfigurationTypeWizard#getGlobalConfigrationElementNode()
+	 */
+	@Override
+	public Element getGlobalConfigurationElementNode() {
+		return inputNode;
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see org.fusesource.ide.camel.editor.provider.ext.
+	 * GlobalConfigurationTypeWizard#setGlobalConfigrationElementNode(org.w3c.
+	 * dom.Element)
+	 */
+	@Override
+	public void setGlobalConfigurationElementNode(Element node) {
+		this.inputNode = node;
+	}
+
+	/**
+	 * @return the component
+	 */
+	public Component getComponent() {
+		Object component = componentManager.getComponentForTag(CamelBean.BEAN_NODE);
+		if (component != null && component instanceof Component) {
+			return (Component) component;
+		}
+		return null;
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see org.eclipse.jface.wizard.Wizard#performFinish()
+	 */
+	@Override
+	public boolean performFinish() {
+		setGlobalConfigurationElementNode(inputNode);
+		return true;
+	}
+
+	/*
+	 * Though this seems nonsensical at this point, it somehow resets the wizard to the point
+	 * where the updated XML element is updated correctly in the model. 
+	 */
+	public void init(IWorkbench workbench, IStructuredSelection selection) {
```

Comment:
Note the comment above this line. I tried removing them, but was right back to things not being persisted properly in the underlying model. So I'm open to suggestions. :) Is there a way for me to safely ignore this comment in this case?